File: app/src/main/java/com/bth/running/adapter/RunAdapter.java
package com.bth.running.adapter;

import android.content.Context;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import com.bth.running.R;
import com.bth.running.running.Run;
import com.bth.running.util.ResourceManager;

import org.joda.time.Period;
import org.joda.time.PeriodType;
import org.joda.time.format.DateTimeFormat;

import java.util.List;

import butterknife.Bind;


/**
 * @author Martin Macheiner
 *         Date: 22.04.2017.
 */

public class RunAdapter extends BaseAdapter<Run> {

    public RunAdapter(Context cxt, List<Run> data) {
        super(cxt, data);
    }

    @Override
    public BaseAdapter<Run>.ViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        return new ViewHolder(inflater.inflate(R.layout.item_run, parent, false));
    }

    @Override
    public void onBindViewHolder(BaseAdapter.ViewHolder holder, int position) {
        holder.bind(data.get(position));
    }

    class ViewHolder extends BaseAdapter<Run>.ViewHolder {

        @Bind(R.id.item_run_date_time)
        TextView txtDatetime;

        @Bind(R.id.item_run_distance)
        TextView txtDistance;

        @Bind(R.id.item_run_duration)
        TextView txtDuration;

        ViewHolder(final View itemView) {
            super(itemView);
        }

        @Override
        public void bind(Run run) {
            content = run;

            txtDuration.setText(ResourceManager.getPeriodFormatter()
                    .print(new Period(run.getTime(), PeriodType.time().withMillisRemoved())));
            txtDatetime.setText(DateTimeFormat.forPattern("dd. MMM yyyy - kk:mm")
                    .print(run.getStartTimeSinceEpoch()));
            txtDistance.setText(ResourceManager.roundDoubleWithDigits(run.getDistance(), 2) + " km");
        }

    }


}

File: app/src/main/java/at/shockbytes/corey/running/adapter/RunAdapter.kt
package at.shockbytes.corey.running.adapter

import android.content.Context
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import at.shockbytes.corey.running.R
import at.shockbytes.corey.running.running.Run
import at.shockbytes.corey.running.util.RunUtils
import at.shockbytes.util.AppUtils
import at.shockbytes.util.adapter.BaseAdapter
import butterknife.BindView
import org.joda.time.Period
import org.joda.time.PeriodType
import org.joda.time.format.DateTimeFormat


/**
 * @author  Martin Macheiner
 * Date:    22.04.2017.
 */

class RunAdapter(cxt: Context, data: List<Run>) : BaseAdapter<Run>(cxt, data.toMutableList()) {

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): BaseAdapter<Run>.ViewHolder {
        return ViewHolder(inflater.inflate(R.layout.item_run, parent, false))
    }

    inner class ViewHolder(itemView: View) : BaseAdapter<Run>.ViewHolder(itemView) {

        @BindView(R.id.item_run_date_time)
        lateinit var txtDatetime: TextView

        @BindView(R.id.item_run_distance)
        lateinit var txtDistance: TextView

        @BindView(R.id.item_run_duration)
        lateinit var txtDuration: TextView

        override fun bind(t: Run) {
            content = t

            txtDuration.text = RunUtils.periodFormatter
                    .print(Period(t.time, PeriodType.time().withMillisRemoved()))
            txtDatetime.text = DateTimeFormat.forPattern("dd. MMM yyyy - kk:mm")
                    .print(t.startTimeSinceEpoch)
            txtDistance.text = AppUtils.roundDouble(t.distance, 2).toString() + " km"
        }

    }


}
